### PR TITLE
Branch appointment test

### LINKED ISSUE #90: Create unit tests for Appointment
The following unit tests have yet to be created (as of 23 Mar 8:30pm)
 
- AppointmentTest
- AppointmentListTest
- AddAppointmentCommandTest
- DeleteAppointmentCommandTest
- EditAppointmentCommandTest
- AddAppointmentCommandParserTest
- DeleteAppointmentCommandParserTest
- EditAppointmentCommandParserTest
- Update ParserUtilTest

### LINKED ISSUE #206: Bugs in edit_appointment command
Editing Name, Date, Time From, or Time To does not work.

Additionally the command format message is incorrect.

### PR DESCRIPTION
**Pull request by jared98lyj**
1. Added tests for appointment and its parser.
2. Fixed bugs for edit appointment and add appointment. Now appointment to be added must ensure that tutor has both Name teaches Subject

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/appointmentcommands/EditAppointmentCommand.java [hunk 1]
```diff
@@ -78,8 +80,15 @@ public CommandResult execute(Model model) throws CommandException {
         Appointment appointmentToEdit = lastShownList.get(index.getZeroBased());
         Appointment editedAppointment = createEditedAppointment(appointmentToEdit, editAppointmentDescriptor);
 
-        if (!appointmentToEdit.equals(editedAppointment) && model.hasAppointment(editedAppointment)) {
+        System.out.println(editedAppointment.getName());
```
Comment (glatiuden) [inline src/main/java/seedu/address/logic/commands/appointmentcommands/EditAppointmentCommand.java]: I think you forgot to remove debugging print statement here

**Review by glatiuden — DISMISSED** [review]
LGTM

**Review by glatiuden — DISMISSED** [review]
LGTM

File: src/main/java/seedu/address/logic/commands/appointmentcommands/EditAppointmentCommand.java (hunk 1 above)
Comment (jared98lyj) [reply src/main/java/seedu/address/logic/commands/appointmentcommands/EditAppointmentCommand.java]: Removed

File: docs/UserGuide.md [hunk 2]
```diff
@@ -358,8 +358,11 @@ Location: Hougang
 ### Editing an appointment: `edit_appointment`
 
 Edits an appointment with a specific index. Only the attributes present are changed in the appointment.
+If any one of the following three fields (DATE, TIME_FROM, TIME_TO), then all three 
+must also be present.
 
-Format: `edit_appointment INDEX [n/NAME] [s/SUBJECT_NAME] [d/DATE] [fr/TIME_FROM] [to/TIME_TO] [l/LOCATION]`
+Format: `edit_appointment [index] [n/name] [s/SUBJECT_NAME] [d/DATE] [fr/TIME_FROM]
```
Comment (kingsleykuan) [inline docs/UserGuide.md]: I think that [index] is incorrect. The square brackets indicate that an argument is optional, which it is not here. Furthermore, index and name should be uppercase

File: src/main/java/seedu/address/logic/parser/appointmentparser/EditAppointmentCommandParser.java [hunk 3]
```diff
@@ -52,21 +58,36 @@ public EditAppointmentCommand parse(String args) throws ParseException {
...
 
-        if (argMultimap.getValue(PREFIX_DATE).isPresent() && argMultimap.getValue(PREFIX_TIME_FROM).isPresent()) {
+        if (argMultimap.getValue(PREFIX_DATE).isPresent()
+                || argMultimap.getValue(PREFIX_TIME_FROM).isPresent()
+                || argMultimap.getValue(PREFIX_TIME_TO).isPresent()) {
+            if (!ArgumentTokenizer.arePrefixesPresent(argMultimap, PREFIX_TIME_FROM,
+                    PREFIX_TIME_TO)) {
+                throw new ParseException(MESSAGE_MISSING_DATE_FIELD);
+            }
+        }
```
Comment (kingsleykuan) [inline src/main/java/seedu/address/logic/parser/appointmentparser/EditAppointmentCommandParser.java]: Why does this only check for PREFIX_TIME_FROM and PREFIX_TIME_TO? What if PREFIX_TIME_FROM is present but PREFIX DATE is missing?

**Review by kingsleykuan — CHANGES_REQUESTED** [review]
Please see comments, ready to merge once those are resolved.

File: src/main/java/seedu/address/logic/parser/appointmentparser/EditAppointmentCommandParser.java (hunk 3 above)
Comment (jared98lyj) [reply src/main/java/seedu/address/logic/parser/appointmentparser/EditAppointmentCommandParser.java]: Fixed

File: docs/UserGuide.md (hunk 2 above)
Comment (kingsleykuan) [reply docs/UserGuide.md]: It should be `INDEX [n/NAME]`

**Comment by kingsleykuan** [comment]
There are also pending merge conflicts to resolve before we can approve!

Also see above for my updated comment

File: docs/UserGuide.md [hunk 4]
```diff
@@ -358,8 +358,11 @@ Location: Hougang
 ### Editing an appointment: `edit_appointment`
 
 Edits an appointment with a specific index. Only the attributes present are changed in the appointment.
+If any one of the following three fields (DATE, TIME_FROM, TIME_TO), then all three 
+must also be present.
 
-Format: `edit_appointment INDEX [n/NAME] [s/SUBJECT_NAME] [d/DATE] [fr/TIME_FROM] [to/TIME_TO] [l/LOCATION]`
+Format: `edit_appointment INDEX [n/name] [s/SUBJECT_NAME] [d/DATE] [fr/TIME_FROM]
```
Comment (kingsleykuan) [inline docs/UserGuide.md]: The argument should be [n/NAME] here

**Review by kingsleykuan — CHANGES_REQUESTED** [review]

**Review by kingsleykuan — APPROVED** [review]
LGTM

**Review by glatiuden — APPROVED** [review]
LGTM

**Pull request merged**